apps/certs/tests/test_services.py

```
@@ -295,3 +295,61 @@ def fake_run(command: list[str], *, env=None):  # noqa: ARG001
+    message = str(exc_info.value)
+    assert "Some challenges have failed" in message
+    assert "Using HTTP-01 webroot" in message
+    assert "/var/log/letsencrypt/letsencrypt.log" in message
+
+
+def test_request_certbot_certificate_godaddy_challenge_failure_includes_dns_hint(
+    monkeypatch, tmp_path
+):
+    """DNS-01 challenge failures should include GoDaddy-specific troubleshooting guidance."""
+
+    def fake_build(*, domain, email, dns_credential, dns_propagation_seconds, dns_use_sandbox, sudo):
```

Ruff (0.15.1)

[warning] Unused function argument: `domain`

(ARG001)